src/test/regress/expected/upgrade_partition_constraints_before.out

```diff
@@ -0,0 +1,28 @@
+-- test cases for #3970
+CREATE SCHEMA test_3970;
+SET search_path = test_3970;
+--1. create a partitioned table
+CREATE TABLE part_table (
+    work_ymdt timestamp without time zone NOT NULL,
+    seq bigint NOT NULL,
+    my_seq bigint NOT NULL,
+    work_memo character varying(150),
+    CONSTRAINT work_memo_check CHECK ((octet_length((work_memo)::text) <= 150))
+)
+PARTITION BY RANGE (work_ymdt);
+--2. perform create_distributed_table
+SELECT create_distributed_table('part_table', 'seq');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+--3. add a partition
+CREATE TABLE part_table_p202008 PARTITION OF part_table FOR VALUES FROM ('2020-08-01 00:00:00') TO ('2020-09-01 00:00:00');
+--4. add a check constraint
```

Comment:
One thing that is not very clear to me is that the same test works on the regular regression test files but not on this file. My questions are probably more related to upgrade testing infra. as opposed to this PR directly.

What is the source Citus version? Do we increase the version on every release or is that static? If we bump the source Citus version to Citus 10, wouldn't this test pass?

Reply:
In citus upgrade tests we currently have 4 versions 8.0, 8.1, 8.2 and 8.3. We only have upgrade tests on pg 11: https://github.com/citusdata/the-process/blob/master/circleci/images/Makefile#L14-L16. Currently it is static and we might need to add some more version there.

For each version, the old version is installed and then the "before" schedule is run. Then the current code is installed and "after" schedule is run. https://github.com/citusdata/citus/tree/master/src/test/regress/upgrade#citus-upgrade-test

>If we bump the source Citus version to Citus 10, wouldn't this test pass?

This is a good point, this test would pass if we had citus 10 in our citus versions array. We could add an alternative output but adding an alternative output is not a good solution in general. 

How useful is it to see the error in the old version? One option is to remove the line that caused the error.